contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -2493,6 +2513,149 @@ is_char_identpart(char c)
 			(c >= '0' && c <= '9'));
 }
 
+/*
+ * Put delimiters around a T-SQL variable/parameter that is
+ * named '@@var' or contains a hash, e.g. '@var#'.
+ * Without delimiters, the backend will raise an error.
+ * This is used for the parameter argument of sp_executesql, so the input
+ * string may contain multiple names, e.g.: @par1 int, @par2 varchar(20), ...
+ */
+static void
+delimit_tsql_atatuservar(const char *src, char *result)
```

Comment:
Why not have this function allocate and return the result itself, instead of requiring a pre-allocated result buffer, which imo has potential for errors if the caller doesn't properly allocate it?

Reply:
What's the preferred way of doing it? There does not seem to be a consistently followed approach as far as I can tell, so I just took inspiration from some existing function.

Reply:
Hm, I'm not sure if we have a standard for this in Babelfish. The decision of whether to accept a pre-allocated buffer or to allocate the memory within the function itself depends on the use case of the function. For example, sometimes the output just needs to be put into a temporary buffer within the caller, and isn't allocated on the heap at all. In those cases it can make sense to allow the caller to be responsible for defining/allocating the output buffer. 

In this case however, your use case so far is using palloc() anyways, so unless you have a strong idea of a future use case of this function that would _not_ be using a palloc() call, I think it's cleaner and safer to have this function allocate the memory itself.

In either case, the behavior with the output needs to be clearly documented in the function header comment. For the current case, that would be specifying that the `result` must have sufficient space to hold the expanded string, and if you change the function to return the result, you should specify that it is returned in a newly palloc'ed string.

Reply:
Changed to return the modified string.